File: gm-utils/outros/sicoob/src/commom/utils/string/StringLength.ts
import IntegerCompare from "../integer/IntegerCompare";
import Null from "../object/Null";

export default class StringLength {

	private constructor() {}

	public static get(s : string) : number {
		if (Null.is(s)) {
			return 0;
		}
		return s.length;
	}

	public static is(s : string, size : number) : boolean {
		return IntegerCompare.eq(StringLength.get(s), size);
	}

	public static max(s : string, max : number) : string {
		if (StringLength.get(s) > max) {
			return s.substring(0, max);
		}
		return s;
	}

}

File: sicoob-proxy/src/main/java/br/utils/strings/StringLength.java
package br.utils.strings;

import br.utils.DevException;

public class StringLength {
	
	public static int get(String s) {
		return s == null ? 0 : s.length();
	}
	
	public static void assertMax(String s, int max) {
		int len = get(s);
		if (len > max) {
			throw DevException.build("length > " + max);
		}
	}
	
	public static String max(String s, int max) {
		int len = get(s);
		if (len > max) {
			s = s.substring(0, max);
		}
		return s;
	}
	
}